wagtail/admin/views/reports/workflows.py

```diff
@@ -188,10 +188,16 @@ def get_queryset(self):
             content_type_id__in=get_editable_content_type_ids(self.request)
         )
 
-        return WorkflowState.objects.filter(editable_pages | editable_objects).order_by(
-            "-created_at"
+        return (
+            WorkflowState.objects.filter(editable_pages | editable_objects)
+            .select_related("workflow", "requested_by")
+            .prefetch_related("content_object")
```

Comment:
q: could this be optimised further by using [`Prefetch`](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.Prefetch)?

https://hakibenita.com/all-you-need-to-know-about-prefetching-in-django#introducing-prefetch is a good read

Reply:
I doubt it, it's a generic foreign key so we don't know the table ahead of time, thus making it harder to add further annotations.

There is still an N+1 query happening on this report page where the template calls `WorkflowState.all_tasks_with_status` on each `WorkflowState`, which we could probably optimise by computing the same logic from the view instead. I tried looking into this but it's quite complex and is way out of scope of this PR as it's been around since even before workflows were introduced to snippets.